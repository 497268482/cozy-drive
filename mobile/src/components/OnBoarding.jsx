import React from 'react'
import classNames from 'classnames'

import { translate } from '../../../src/lib/I18n'
import Breadcrumb from './Breadcrumb'
import styles from '../styles/onboarding'

export const OnBoarding = ({ t, onActivate, onSkip, nameStep, currentStep, totalSteps }) =>
(
  <div className={classNames(styles['wizard'], styles['photos-backup'])}>
    <header className={styles['wizard-header']}>
      {onSkip && <a className={styles['skipLink']} onClick={onSkip}>
        {t('mobile.onboarding.step.skip')}
      </a>}
    </header>
    <div className={styles['wizard-main']}>
      <div className={classNames(styles['illustration'], styles['illustration-' + nameStep])} />
      <h1 className={styles['title']}>{t('mobile.onboarding.' + nameStep + '.title')}</h1>
      <p className={styles['description']}>{t('mobile.onboarding.' + nameStep + '.description')}</p>
    </div>
    <footer className={styles['wizard-footer']}>
      <button
        role='button'
        className={'coz-btn coz-btn--regular'}
        onClick={onActivate}
      >
        {onSkip && t('mobile.onboarding.step.button')}
        {!onSkip && t('mobile.onboarding.step.next')}
      </button>
      <Breadcrumb currentStep={currentStep} totalSteps={totalSteps} />
    </footer>
  </div>
)

export default translate()(OnBoarding)
