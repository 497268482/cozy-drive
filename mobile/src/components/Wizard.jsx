import React from 'react'

export default class Wizard extends React.Component {
  constructor (props) {
    super(props)

    this.nextStep = this.nextStep.bind(this)

    this.state = {
      step: 0
    }
  }

  nextStep () {
    this.setState({ step: this.state.step + 1 })
  }

  render () {
    const Step = this.props.steps[this.state.step]
    return <Step nextStep={this.nextStep} {...this.props} />
  }
}
