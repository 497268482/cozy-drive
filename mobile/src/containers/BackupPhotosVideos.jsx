import React from 'react'
import { connect } from 'react-redux'
import classnames from 'classnames'
import styles from '../styles/onboarding.styl'

import logo from '../../res/icon.png'

export const BackupPhotosVideos = ({ t, onActivate, onSkip }) =>
(
  <div className={classnames(styles['wizard'], styles['activation'])}>
    <div className={classnames(styles['wizard-main'])}>
      <img src={logo} alt='logo' />
      <h1>{t('mobile.onboarding.activation.backupPhotosVideos.title')}</h1>
      <p>{t('mobile.onboarding.activation.backupPhotosVideos.description')}</p>
    </div>
    <button role='button' className={classnames('coz-btn coz-btn--regular', styles['wizard-button'])} onClick={onActivate}>{t('mobile.onboarding.activation.button')}</button>
    <a onClick={onSkip}>{t('mobile.onboarding.activation.skip')}</a>
    <div className={classnames(styles['breadcrumb'])}>
      <span>1</span>
      <span>2</span>
      <span>3</span>
      <span>4</span>
    </div>
  </div>
)

const mapStateToProps = (state, ownProps) => ({})

const mapDispatchToProps = (dispatch, ownProps) => ({
  onActivate: () => {
    dispatch({ type: 'UPDATE_SETTINGS', newSettings: { backupImages: true } })
    ownProps.nextStep()
  },
  onSkip: () => {
    dispatch({ type: 'UPDATE_SETTINGS', newSettings: { backupImages: false } })
    ownProps.nextStep()
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(BackupPhotosVideos)
