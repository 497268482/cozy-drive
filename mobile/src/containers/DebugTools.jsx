import React, { Component } from 'react'
import { connect } from 'react-redux'
import { logException, logInfo } from '../lib/reporter'
import { setFirstReplication, setOffline } from '../../../src/actions'

const Button = ({ onClick, children }) => (
  <button onclick={onClick} className={'coz-btn coz-btn--regular'}>
    {children}
  </button>
)

const Checkbox = ({ onChange, value, title }) => (
  <div>
    {title} <input type='checkbox' checked={value} onChange={onChange} />
  </div>
)

class DebugTools extends Component {
  sendSentryException () {
    logException('a debug exception')
  }

  sendSentryMessage () {
    logInfo('a debug message')
  }

  render () {
    return (
      <div>
        <h4>Sentry</h4>
        <Button onClick={() => this.sendSentryException()}>send exception</Button>
        <Button onClick={() => this.sendSentryMessage()}>send message</Button>
        <h4>Offline</h4>
        <Checkbox title='First Replication' value={this.props.firstReplication} onChange={this.props.setFirstReplication} />
        <Checkbox title='Offline' value={this.props.offline} onChange={this.props.setOffline} />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  setFirstReplication: (e) => dispatch(setFirstReplication(e.target.checked)),
  setOffline: (e) => dispatch(setOffline(e.target.checked))
})

const mapStateToProps = (state) => ({
  firstReplication: state.settings.firstReplication,
  offline: state.settings.offline
})

export default connect(mapStateToProps, mapDispatchToProps)(DebugTools)
