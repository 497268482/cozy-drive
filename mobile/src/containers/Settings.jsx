import React from 'react'
import { connect } from 'react-redux'
import { translate } from '../../../src/lib/I18n'

import styles from '../styles/settings'

const Settings = ({t, version, serverUrl}) => (
  <div className={styles['settings']}>
    <h3 className={styles['settings__category']}>{t('mobile.settings.about')}</h3>
    <h4 className={styles['settings__subcategory']}>{t('mobile.settings.app_version')}</h4>
    <p className={styles['settings__item']}>{version}</p>
    <h4 className={styles['settings__subcategory']}>{t('mobile.settings.account')}</h4>
    <p className={styles['settings__item']}>{serverUrl}</p>
  </div>
)

const mapStateToProps = (state, ownProps) => ({
  version: window.navigator.appInfo ? window.navigator.appInfo.version : 'dev (run in cordova to get the config.xml version)',
  serverUrl: state.mobile.serverUrl
})

export default translate()(connect(mapStateToProps)(Settings))
