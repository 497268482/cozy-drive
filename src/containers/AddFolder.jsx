import React from 'react'
import { connect } from 'react-redux'
import classNames from 'classnames'

import styles from '../styles/table'

import { createFolder } from '../actions'
import FilenameInput from '../components/FilenameInput'

const AddFolder = ({ visible, submitting, onSubmit }) => {
  if (!visible) {
    return null
  }
  return (
    <tr>
      <td class={classNames(styles['fil-content-file'], styles['fil-file-folder'])}>
        <FilenameInput submitting={submitting} onSubmit={onSubmit} />
      </td>
      <td>
        
      </td>
      <td>—</td>
      <td>—</td>
    </tr>
  )
}

const mapStateToProps = (state, ownProps) => ({
  visible: state.ui.isAddingFolder,
  submitting: state.ui.isWorking
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  onSubmit: (name) => dispatch(createFolder(name))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddFolder)
