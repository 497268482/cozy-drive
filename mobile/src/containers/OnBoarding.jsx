import React from 'react'

import Wizard from '../components/Wizard'
import Welcome from './onboarding/Welcome'
import SelectServer from './onboarding/SelectServer'
import BackupPhotosVideos from './onboarding/BackupPhotosVideos'

const OnBoarding = (props) => {
  const steps = [
    Welcome,
    SelectServer,
    BackupPhotosVideos
  ]
  return <Wizard steps={steps} {...props} />
}

export default OnBoarding
