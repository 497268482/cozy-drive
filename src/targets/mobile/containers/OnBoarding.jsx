import React from 'react'

import Wizard from '../components/Wizard'
import Welcome from './onboarding/Welcome'
import SelectServer from './onboarding/SelectServer'
import Files from './onboarding/Files'
import BackupPhotosVideos from './onboarding/BackupPhotosVideos'
import Analytics from './onboarding/Analytics'

const OnBoarding = (props) => {
  const steps = [
    Welcome,
    SelectServer,
    Files,
    BackupPhotosVideos,
    Analytics
  ]
  return <Wizard steps={steps} {...props} />
}

export default OnBoarding
