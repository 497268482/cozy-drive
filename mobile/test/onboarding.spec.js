'use strict'

import React from 'react'
import renderer from 'react-test-renderer'

import Wizard from '../../mobile/src/components/Wizard'
import { Welcome, SelectServer } from '../../mobile/src/containers/OnBoarding'

describe('Wizard Component', () => {
  it('should render different components', () => {
    const steps = [
      Welcome,
      SelectServer
    ]
    const component = renderer.create(
      <Wizard steps={steps} t={() => {}} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    tree.children[1].props.onClick()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
