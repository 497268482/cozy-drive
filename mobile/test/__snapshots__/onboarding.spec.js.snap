exports[`Onboarding should render different components 1`] = `
<div
  className="wizard welcome">
  <div
    className="wizard-main">
    <img
      alt="logo"
      src="test-file-stub" />
    <h1 />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={[Function]}
    role="button" />
</div>
`;

exports[`Onboarding should render different components 2`] = `
<div
  className="wizard">
  <div
    className="wizard-main">
    <p />
    <input
      className="http-input"
      onChange={undefined}
      pattern="https://.+"
      placeholder={undefined}
      type="url"
      value={undefined} />
    <p />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={undefined}
    role="button" />
</div>
`;

exports[`Wizard Component should render the SelectServer screen 1`] = `
<div
  className="wizard">
  <div
    className="wizard-main">
    <p />
    <input
      className="http-input"
      onChange={undefined}
      pattern="https://.+"
      placeholder={undefined}
      type="url"
      value={undefined} />
    <p />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={undefined}
    role="button" />
</div>
`;

exports[`Wizard Component should render the welcome screen 1`] = `
<div
  className="wizard welcome">
  <div
    className="wizard-main">
    <img
      alt="logo"
      src="test-file-stub" />
    <h1 />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={undefined}
    role="button" />
</div>
`;
